src/components/dataset/data/sidebar/filter/CategoryWrapper.test.tsx

```diff
@@ -0,0 +1,189 @@
+import history from '../../../../../modules/hist';
+import { ActionTypes } from '../../../../../constants';
+
+function createState(column: TableColumnType): any {
+  return {
+    datasets: {
+      dataset: {
+        description: 'A dataset description',
+        phsId: '12345',
+      },
+    },
+    column,
+  };
+}
+
+const baseStringColumn = {
+  name: 'variant_id',
+  label: 'variant_id',
+};
+
+let store: MockStoreEnhanced<unknown, unknown>;
+
+function mountComponent(state: any) {
+  const mockStore = createMockStore([routerMiddleware(history)]);
+  store = mockStore(state);
+  mount(
+    <Provider store={store}>
+      <Router history={history}>
+        <ThemeProvider theme={globalTheme}>
+          <CategoryWrapper
+            key={state.column.name}
+            column={state.column}
+            filterMap={{}}
+            handleChange={() => null}
+            handleFilters={() => null}
+            tableName="tableName"
+            toggleExclude={() => null}
+          />
+        </ThemeProvider>
+      </Router>
+    </Provider>,
+  );
+}
+
+describe('Initial load of CategoryWrapper', () => {
+  beforeEach(() => {
+    const initialState = createState({ ...baseStringColumn, isExpanded: false });
+    mountComponent(initialState);
+  });
+  it('should have NOT loaded column stats', () => {
+    assertGetColumnStatsWasNotCalled();
+  });
+  it('No values are returned, so component should not be displayed', () => {
+    cy.get('[data-cy="categoryFilterCheckbox-a"]').should('not.exist');
+    cy.get('[id="autocomplete-variant_id"]').should('not.exist');
+    cy.get('circle').should('not.exist');
+  });
+});
+
+describe('First expansion of column', () => {
+  beforeEach(() => {
+    const state = createState({ ...baseStringColumn, isExpanded: true });
+    mountComponent(state);
+  });
+  it('Load column stats on first load', () => {
+    assertGetColumnStatsWasCalled();
+  });
+});
+
+describe('Expanded column after another column has updated - Checkbox', () => {
+  beforeEach(() => {
+    const state = createState({
+      ...baseStringColumn,
+      isExpanded: true,
+      filterHasUpdated: true,
+      values: [{ value: 'a', count: 1 }],
+      originalValues: [
+        { value: 'a', count: 1 },
+        { value: 'b', count: 2 },
+      ],
+    });
+    mountComponent(state);
+  });
+  it('Load column stats after filter has changed for checkbox fields', () => {
+    assertGetColumnStatsWasCalled();
+  });
+  it('Checkbox component is displayed', () => {
+    cy.get('[data-cy="categoryFilterCheckbox-a"]').should('exist');
+  });
+});
+
+describe('Expanded column after another column has updated - Freetext', () => {
+  beforeEach(() => {
+    const state = createState({
+      ...baseStringColumn,
+      isExpanded: true,
+      filterHasUpdated: true,
+      values: [{ value: 'a', count: 1 }],
+      originalValues: [
+        { value: 'a', count: 1 },
+        { value: 'b', count: 2 },
```

Comment:
I'm not sure I understand, why does this test contain repeated values?

Reply:
We don't have anything on the UI side that checks if the values are unique (We expect the API to return a list of unique values). For this test, I'm trying to just get more than 30 elements in the values list so that it switch from the checkbox display to the freetext dropdown. But, it is unclear- I'll make them unique.

Reply:
Oh that makes sense now!